python/src/aac/lang/definitions/metadata_structure.py

```diff
@@ -0,0 +1,83 @@
+"""This module provides functions to collate and build a meta structure for an AaC definition."""
+from copy import deepcopy
+from typing import Any, Optional
+from aac.lang.definitions.definition import Definition
+from aac.lang.definitions.lexeme import Lexeme
+from aac.lang.definitions.schema import get_root_schema_definitions, get_schema_for_field
+from aac.lang.definitions.structural_node import NodeType, StructuralNode
+from aac.lang.language_context import LanguageContext
+
+
+def compute_metadata_structure(definition: Definition, language_context: LanguageContext) -> None:
+    """
+    Builds the metadata structure for the definition and attaches it the passed-in definition's metastructure attribute.
+
+    Args:
+        definition (Definition): The definition to compute a metastructure for
+        language_context (LanguageContext): The language context used to provide the metadata
+    """
+    token_traversal_list = definition.lexemes.copy()  # Create a copy to avoid altering the referenced list
+    yaml_traversal_structure = deepcopy(definition.structure)  # Create a copy to avoid altering the referenced yaml structure
+
+    root_token = token_traversal_list.pop(0)
+    root_definition = get_root_schema_definitions(language_context).get(root_token.value)
+
+    root_node = StructuralNode(NodeType.ROOT_KEY, root_definition, root_token, None, [])
+    root_yaml_structure = yaml_traversal_structure.get(root_token.value, {})
+    _recursive_meta_structure_build(
+        definition, token_traversal_list, root_yaml_structure, root_node, [root_token.value], language_context
+    )
+
+    definition.meta_structure = root_node
+
+
+def _recursive_meta_structure_build(
+    definition: Definition,
+    tokens: list[Lexeme],
+    yaml_structure: dict[str, Any],
+    parent_node: StructuralNode,
+    traversed_keys: list[str],
+    language_context: LanguageContext,
+) -> None:
+    """Recursively builds the meta structure nodes and attaches them to the parent node."""
+
+    child_nodes = []
+
+    def _create_node(
+        node_type: NodeType, node_token: Optional[Lexeme] = None, defining_definition: Optional[Definition] = None
```

Comment:
Would follow similar style as above and have each parameter be on a line since there are multiple, especially since it is already split from the method name.

Reply:
This is formatted based using our [auto-formatter](https://black.readthedocs.io/en/stable/).

Reply:
Ah, so guessing that has a threshold for swapping to single line entries then and this hadn't met it.

Reply:
Yeah, it's a few characters shy of taking it to the extreme of one-line per parameter.